src/frontend/src/lib/utils/qr-code.utils.ts

```diff
@@ -1,7 +1,49 @@
-import type { QrResponse, QrStatus } from '$lib/types/qr-code';
+import {
+	urnNumericParams,
+	urnStringParams,
+	type DecodedUrn,
+	type QrResponse,
+	type QrStatus
+} from '$lib/types/qr-code';
 import { decodePayment } from '@dfinity/ledger-icrc';
 import { isNullish, nonNullish, type Token } from '@dfinity/utils';
 
+export const decodeUrn = (urn: string): DecodedUrn | undefined => {
+	const regex = /^([a-zA-Z]+):([a-zA-Z0-9\-.]+)(@(\d+))?(\/([a-zA-Z]+))?(\?(.*))?$/;
```

Comment:
Can you add some documentation about the source of the regex?

Reply:
on it!